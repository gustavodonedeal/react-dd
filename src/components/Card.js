import React from 'react';
import { Link } from 'react-router-dom';
import MediaPreview from './MediaPreview';
import PriceTag from './PriceTag';

class Card extends React.Component{

  render(){
    const ad = this.props.ad;

    return (
        <li className="card-item">
          <Link className="card__link" to={`/ad/${ad.id}`}>
            <div className="card">
              <MediaPreview ad={ad} />
              <div className="card__body">
                <div className="card__body-top">
                  <p className="card__body-title">{ad.header}</p>
                  <ul className="card__body-keyinfo">
                    <li>{ad.age}</li>
                    <li>{ad.county}</li>
                  </ul>
                </div>
                <div className="card__body-lower">
                  <PriceTag ad={ad} />
                  <div className="card__actions">
                    <i className="manual-false-page card-actions__save icon icon-save-heart"></i>
                  </div>
                </div>
              </div>
            </div>
          </Link>
        </li>
    )
  }
}

export default Card;
