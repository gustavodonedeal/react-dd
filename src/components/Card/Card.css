.card-item {
  margin-bottom: 14px;
}
.card__link,
.card__link:hover {
  text-decoration: none;
}

.card{
  display: flex;
  position: relative;
  background: #fff;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, .2);
  border-radius: 2px;
  overflow: hidden;
  text-align: left;
}

.card__media{
  min-width: 200px;
  -webkit-box-flex: 1;
  flex: 1 1 200px;
  max-width: 300px;
  position: relative;
}

.card__photo{
  background-color: #e0e0db;
  position: relative;
  display: block;
  width: 100%;
  height: 0;
  padding: 75% 0 0;
  overflow: hidden;
}

.card__photo img{
  position: absolute;
  display: block;
  max-width: 100%;
  max-height: 100%;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  margin: auto;
}

.card__media-count{
  position: absolute;
  left: 10px;
  bottom: 10px;
  padding: 8px;
  background: rgba(0, 0, 0, .4);
  border-radius: 4px;
  color: hsla(0, 0%, 100%, .6);
  font-size: 13px;
  font-weight: 400;
}

.card__media-count .icon-camera{
  display: inline-block;
  background-size: contain;
  height: 19px;
  width: 19px;
  margin: -4px 4px -4px 0;
}

.icon-camera{
  background-image: url(data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%20version%3D%221.1%22%20id%3D%22camera-Layer_1%22%20x%3D%220px%22%20y%3D%220px%22%20viewBox%3D%220%200%2013.4%2010.3%22%20style%3D%22enable-background%3Anew%200%200%2013.4%2010.3%3B%22%20xml%3Aspace%3D%22preserve%22%20width%3D%2232%22%20height%3D%2232%22%3E%3Cstyle%20type%3D%22text%2Fcss%22%3E%09.st0%7Bopacity%3A0.8%3B%7D%09.st1%7Bfill%3A%23FFFFFF%3B%7D%3C%2Fstyle%3E%3Cg%20id%3D%22camera-XsEodM.tif_1_%22%20class%3D%22st0%22%3E%09%3Cg%3E%09%09%3Cg%3E%09%09%09%3Cpath%20class%3D%22st1%22%20d%3D%22M3.7%2C1.8C3.8%2C0.9%2C4.3%2C0.3%2C5%2C0.1C5.2%2C0%2C5.4%2C0%2C5.5%2C0C6.3%2C0%2C7%2C0%2C7.7%2C0c1%2C0%2C1.9%2C0.6%2C1.9%2C1.8c0.1%2C0%2C0.1%2C0%2C0.2%2C0%20%20%20%20%20c0.6%2C0%2C1.2%2C0%2C1.8%2C0c1%2C0%2C1.8%2C0.8%2C1.8%2C1.8c0%2C1.7%2C0%2C3.3%2C0%2C5c0%2C1-0.8%2C1.8-1.8%2C1.8c-3.3%2C0-6.6%2C0-9.9%2C0c-1%2C0-1.7-0.8-1.7-1.8%20%20%20%20%20c0-1.7%2C0-3.4%2C0-5c0-1%2C0.7-1.8%2C1.7-1.8c0.6%2C0%2C1.2%2C0%2C1.8%2C0C3.6%2C1.8%2C3.7%2C1.8%2C3.7%2C1.8z%20M6.7%2C3C5%2C3%2C3.6%2C4.3%2C3.6%2C6%20%20%20%20%20c0%2C1.7%2C1.4%2C3.1%2C3.1%2C3.1c1.7%2C0%2C3.1-1.4%2C3.1-3.1C9.8%2C4.3%2C8.4%2C3%2C6.7%2C3z%22%2F%3E%09%09%09%3Cpath%20class%3D%22st1%22%20d%3D%22M6.7%2C7.8C5.7%2C7.8%2C5%2C7%2C5%2C6c0-1%2C0.8-1.7%2C1.7-1.7c1%2C0%2C1.7%2C0.8%2C1.7%2C1.7C8.4%2C7%2C7.7%2C7.8%2C6.7%2C7.8z%22%2F%3E%09%09%3C%2Fg%3E%09%3C%2Fg%3E%3C%2Fg%3E%3C%2Fsvg%3E);
  background-repeat: no-repeat;
}

.card__body{
  min-width: 280px;
  -webkit-box-flex: 2;
  flex: 2 1 570px;
  max-width: none;
  padding: 12px 10px 12px 20px;
  display: flex;
  -webkit-box-direction: normal;
  -webkit-box-orient: vertical;
  flex-direction: column;
  -webkit-box-pack: justify;
  justify-content: space-between;
}

.card__body-title{
  font-size: 18px;
  color: rgba(0, 0, 0, .75);
  margin-bottom: 0;
}