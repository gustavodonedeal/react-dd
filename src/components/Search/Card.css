:local .cardContainer {
  display: grid;
  grid-template-columns: 2fr 1fr 1fr;
  grid-template-rows: repeat(3, 1fr);
  grid-gap: 10px;
  grid-template-areas: 
    "image header header" 
    "image info info"
    "image price price";
}

:local .cardImage {
  grid-area: image;
  width: 100%;
  height: 100%;
  object-fit: cover;
}

:local .cardHeader {
  grid-area: header;
}

:local .cardInfo {
  grid-area: info;
}

:local .price {
  grid-area: price;
}