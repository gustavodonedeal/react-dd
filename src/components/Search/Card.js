import React from 'react';
import cardStyles from './Card.css';
import { Link } from 'react-router-dom';

const CurrencySign = ({ currency }) => {
  switch (currency) {
    case 'EUR':
    default:
      return <span>â‚¬</span>;
  }
};

const Price = ({ currency, children }) => (
  <div>
    <CurrencySign currency={currency} /> {children}
  </div>
);

const Card = ({ header, age, county, price, currency, photos, id }) => (
  <Link to={`/ad/${id}`} className={cardStyles.cardContainer}>
    <img src={photos ? photos[0].large : ''} className={cardStyles.cardImage} />
    <h3 className={cardStyles.cardHeader}>{header}</h3>
    <div className={cardStyles.cardInfo}>{`${age} | ${county}`}</div>
    <h4 className={cardStyles.price}>
      <Price currency={currency}>{price}</Price>
    </h4>
  </Link>
);

export default Card;
