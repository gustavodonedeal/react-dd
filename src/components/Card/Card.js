import React from 'react';
import { Link } from 'react-router-dom';
import MediaPreview from '../MediaPreview';
import PriceTag from '../PriceTag';
import './Card.css';

class Card extends React.Component{

  render(){
    const ad = this.props.ad;
    const photo = (ad.photos && ad.photos.length > 0 ? ad.photos[0].small2 : null);

    return (
      <li className="card-item">
        <Link className="card__link" to={`/ad/${ad.id}`}>
          <div className="card">
            <MediaPreview photo={photo} mediaCount={ad.mediaCount} header={ad.header} />
            <div className="card__body">
              <div className="card__body-top">
                <p className="card__body-title">{ad.header}</p>
                <ul className="card__body-keyinfo">
                  <li>{ad.age}</li>
                  <li>{ad.county}</li>
                </ul>
              </div>
              <div className="card__body-lower">
                <PriceTag price={ad.price} currency={ad.currency} isWantedAd={ad.wanted} />
                <div className="card__actions">
                  <i className="manual-false-page card-actions__save icon icon-save-heart" />
                </div>
              </div>
            </div>
          </div>
        </Link>
      </li>
    )
  }
}

export default Card;
