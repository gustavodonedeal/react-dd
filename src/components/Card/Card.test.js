import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import Card from './Card';

it('renders inital loading state correctly', async () => {
  const json = await import('../../../__tests__/sampleAdvert.json');
  const immutableAdvert = Object.freeze(json);
  const tree = renderer.create(
    <MemoryRouter>
      <Card ad={immutableAdvert} />
    </MemoryRouter>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
