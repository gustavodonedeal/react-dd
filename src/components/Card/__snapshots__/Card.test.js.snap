// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders inital loading state correctly 1`] = `
<li
  className="card-item"
>
  <a
    className="card__link"
    href="/ad/16140864"
    onClick={[Function]}
  >
    <div
      className="card"
    >
      <div
        className="card__media"
      >
        <div
          className="card__photo"
        >
          <img
            alt="Ford Focus 1.6 Automatic"
            src="https://donedeal.akamaized.net/ddimg/N2JiZmM0NGYyNmUzYWQxMGJiODRjYzJhMDAyOD…vbmVkZWFsLmllLXBob3Rvcy9waG90b183ODAwNzA3NHx8fHx8fDYwMHg0NTB8fHx8fA==.jpeg"
          />
        </div>
        <div
          className="card__media-count"
        >
          <i
            className="icon-camera"
          />
          1/
          6
        </div>
      </div>
      <div
        className="card__body"
      >
        <div
          className="card__body-top"
        >
          <p
            className="card__body-title"
          >
            Ford Focus 1.6 Automatic
          </p>
          <ul
            className="card__body-keyinfo"
          >
            <li>
              2 hours
            </li>
            <li>
              Louth
            </li>
          </ul>
        </div>
        <div
          className="card__body-lower"
        >
          <p
            className="card__price"
          >
            <span>
              <span>
                €
              </span>
              2,600
            </span>
          </p>
          <div
            className="card__actions"
          >
            <i
              className="manual-false-page card-actions__save icon icon-save-heart"
            />
          </div>
        </div>
      </div>
    </div>
  </a>
</li>
`;
